tests/unit/test_results.py

```diff
@@ -0,0 +1,170 @@
+import copy
+import datetime
+import random
+
+import responses
+
+from intezer_sdk.analyses_history import AnalysesHistory
+from intezer_sdk.api import get_global_api
+from intezer_sdk.results import Results
+from tests.unit.base_test import BaseTest
+
+BASE_DATA = AnalysesHistory._data_analyses_history(
+    start_date=datetime.datetime.now() - datetime.timedelta(days=3),
+    end_date=datetime.datetime.now()
+)
+
+NO_RESULTS = {'total_count': 0, 'analyses': []}
+NORMAL_RESULT = {
+    'total_count': 2,
+    'analyses': [{'account_id': '123'}, {'account_id': '456'}]
+}
+
+FILE_ANALYSES_REQUEST = '/analyses/history'
+URL_ANALYSES_REQUEST = '/url-analyses/history'
+ENDPOINT_ANALYSES_REQUEST = '/endpoint-analyses/history'
+
+
+class ResultsSpec(BaseTest):
+    def test_fetch_page_no_result(self):
+        """
+        When got no results expect to raise StopIteration exception for stopping
+        the iteration that is going on outer scope.
+        """
+        with responses.RequestsMock() as mock:
+            mock.add('POST',
+                     url=self.full_url + FILE_ANALYSES_REQUEST,
+                     status=200,
+                     json=NO_RESULTS)
+            results = Results(FILE_ANALYSES_REQUEST, get_global_api(), BASE_DATA)
+
+            with self.assertRaises(StopIteration):
+                results._fetch_page()
```

Comment:
we are rarely test private methods, I don't see a reason to do it here

Reply:
I think because its a core function of the class. Do you think this test is unneccessary?